### PR TITLE
[14.0] [MIG] mail_tracking_mass_mailing: MIgration to v14.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT30287 

please @victoralmau @joao-p-marques review this please

### CODE REVIEW

File: mail_tracking_mass_mailing/models/mail_mail.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from odoo import api, models
+
+
+class MailMail(models.Model):
+    _inherit = "mail.mail"
+
+    @api.model
+    def _tracking_email_prepare(self, partner, email):
+        res = super(MailMail, self)._tracking_email_prepare(partner, email)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_mail.py]:
```suggestion
        res = super()._tracking_email_prepare(partner, email)
```

File: mail_tracking_mass_mailing/models/mail_tracking_email.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    mail_id_int = fields.Integer(string="Mail ID", readonly=True)
+
+    @api.model
+    def _statistics_link_prepare(self, tracking):
+        """Inherit this method to link other object to mailing.trace"""
+        return {"mail_tracking_id": tracking.id}
+
+    @api.model
+    def create(self, vals):
+        tracking = super(MailTrackingEmail, self).create(vals)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_email.py]:
```suggestion
        tracking = super().create(vals)
```

File: mail_tracking_mass_mailing/models/mail_tracking_email.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            recipients.append(event.recipient_address)
+        else:
+            recipients = list(filter(None, self.mapped("recipient_address")))
+        for recipient in recipients:
+            self.env["mailing.contact"].search(
+                [("email", "=ilike", recipient)]
+            ).email_bounced_set(self, reason, event=event)
+
+    def smtp_error(self, mail_server, smtp_server, exception):
+        res = super(MailTrackingEmail, self).smtp_error(
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_email.py]:
```suggestion
        res = super().smtp_error(
```

File: mail_tracking_mass_mailing/models/mail_tracking_email.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            mail_server, smtp_server, exception
+        )
+        self._contacts_email_bounced_set("error")
+        return res
+
+    def event_create(self, event_type, metadata):
+        res = super(MailTrackingEmail, self).event_create(event_type, metadata)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_email.py]:
```suggestion
        res = super().event_create(event_type, metadata)
```

File: mail_tracking_mass_mailing/models/mail_tracking_event.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        string="Mass mailing",
+        comodel_name="mailing.mailing",
+        readonly=True,
+        related="tracking_email_id.mass_mailing_id",
+        store=True,
+    )
+
+    @api.model
+    def process_open(self, tracking_email, metadata):
+        res = super(MailTrackingEvent, self).process_open(tracking_email, metadata)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_event.py]:
```suggestion
        res = super().process_open(tracking_email, metadata)
```

File: mail_tracking_mass_mailing/models/mail_tracking_event.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        mail_mail_stats.set_opened(mail_mail_ids=[tracking_email.mail_id_int])
+        return res
+
+    def _tracking_set_bounce(self, tracking_email, metadata):
+        mail_mail_stats = self.sudo().env["mailing.trace"]
+        mail_mail_stats.set_bounced(mail_mail_ids=[tracking_email.mail_id_int])
+
+    @api.model
+    def process_hard_bounce(self, tracking_email, metadata):
+        res = super(MailTrackingEvent, self).process_hard_bounce(
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_event.py]:
```suggestion
        res = super().process_hard_bounce(
```

File: mail_tracking_mass_mailing/models/mail_tracking_event.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            tracking_email, metadata
+        )
+        self._tracking_set_bounce(tracking_email, metadata)
+        return res
+
+    @api.model
+    def process_soft_bounce(self, tracking_email, metadata):
+        res = super(MailTrackingEvent, self).process_soft_bounce(
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_event.py]:
```suggestion
        res = super().process_soft_bounce(
```

File: mail_tracking_mass_mailing/models/mail_tracking_event.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            tracking_email, metadata
+        )
+        self._tracking_set_bounce(tracking_email, metadata)
+        return res
+
+    @api.model
+    def process_reject(self, tracking_email, metadata):
+        res = super(MailTrackingEvent, self).process_reject(tracking_email, metadata)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_event.py]:
```suggestion
        res = super().process_reject(tracking_email, metadata)
```

File: mail_tracking_mass_mailing/models/mail_tracking_event.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        self._tracking_set_bounce(tracking_email, metadata)
+        return res
+
+    @api.model
+    def process_spam(self, tracking_email, metadata):
+        res = super(MailTrackingEvent, self).process_spam(tracking_email, metadata)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/models/mail_tracking_event.py]:
```suggestion
        res = super().process_spam(tracking_email, metadata)
```

File: mail_tracking_mass_mailing/tests/test_mass_mailing.py [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+from odoo.tests.common import TransactionCase, tagged
+from odoo.tools import mute_logger
+
+mock_send_email = "odoo.addons.base.models.ir_mail_server." "IrMailServer.send_email"
+
+
+@tagged("-at_install", "post_install")
+class TestMassMailing(TransactionCase):
+    def setUp(self, *args, **kwargs):
+        super(TestMassMailing, self).setUp(*args, **kwargs)
```
Comment (victoralmau) [inline mail_tracking_mass_mailing/tests/test_mass_mailing.py]:
```suggestion
        super().setUp(*args, **kwargs)
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Good job, code review ok.
Please, remove contributors email from all files.

**Comment by CarlosRoca13** [comment]
@victoralmau changes done

**Review by victoralmau — APPROVED** [review]

File: mail_tracking_mass_mailing/hooks.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    from odoo.addons.mail_tracking.hooks import column_add_with_value
+except ImportError:
+    column_add_with_value = False
+
+_logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(cr):
+    if column_add_with_value:
+        _logger.info("Creating mailing_contact.email_score column " "with value 50.0")
```
Comment (joao-p-marques) [inline mail_tracking_mass_mailing/hooks.py]:
```suggestion
        _logger.info("Creating mailing_contact.email_score column with value 50.0")
```

File: mail_tracking_mass_mailing/tests/test_mass_mailing.py [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from unittest import mock
+
+from odoo.tests.common import TransactionCase, tagged
+from odoo.tools import mute_logger
+
+mock_send_email = "odoo.addons.base.models.ir_mail_server." "IrMailServer.send_email"
```
Comment (joao-p-marques) [inline mail_tracking_mass_mailing/tests/test_mass_mailing.py]:
```suggestion
mock_send_email = "odoo.addons.base.models.ir_mail_server.IrMailServer.send_email"
```

**Review by joao-p-marques — APPROVED** [review]
Code review :+1: Just a couple of minor aesthetic comments, but LGTM

**Comment by CarlosRoca13** [comment]
@joao-p-marques changes done

**Review by joao-p-marques — APPROVED** [review]

**Pull request merged**